tethys_apps/migrations/0004_auto_20230223_1731.py

```diff
@@ -0,0 +1,107 @@
+# Generated by Django 3.2.16 on 2023-02-23 17:31
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("tethys_apps", "0003_alter_tethysapp_feedback_emails"),
+    ]
+
+    operations = [
+        ## Rename Model
+        migrations.RenameModel("customsetting", "oldcustomsetting"),
+        migrations.CreateModel(
+            name="CustomSettingBase",
+            fields=[
+                (
+                    "tethysappsetting_ptr",
+                    models.OneToOneField(
+                        auto_created=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        parent_link=True,
+                        primary_key=True,
+                        serialize=False,
+                        to="tethys_apps.tethysappsetting",
+                    ),
+                ),
+                (
+                    "type_custom_setting",
+                    models.CharField(blank=True, default="", max_length=1024),
+                ),
+            ],
+            bases=("tethys_apps.tethysappsetting",),
+        ),
+        migrations.CreateModel(
+            name="JSONCustomSetting",
+            fields=[
+                (
+                    "customsettingbase_ptr",
+                    models.OneToOneField(
+                        auto_created=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        parent_link=True,
+                        primary_key=True,
+                        serialize=False,
+                        to="tethys_apps.customsettingbase",
+                    ),
+                ),
+                ("value", models.JSONField(blank=True, default=dict)),
+                ("default", models.JSONField(blank=True, default=dict)),
+            ],
+            bases=("tethys_apps.customsettingbase",),
+        ),
+        migrations.CreateModel(
+            name="SecretCustomSetting",
+            fields=[
+                (
+                    "customsettingbase_ptr",
+                    models.OneToOneField(
+                        auto_created=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        parent_link=True,
+                        primary_key=True,
+                        serialize=False,
+                        to="tethys_apps.customsettingbase",
+                    ),
+                ),
+                ("value", models.CharField(blank=True, default="", max_length=1024)),
+            ],
+            bases=("tethys_apps.customsettingbase",),
+        ),
+        # create new CustomSetting
+        migrations.CreateModel(
```

Comment:
Have you tested the migration now that we made the changes to the names? The table for this model will exist, so creating it will probably fail. It needs to modify it instead. Probably would be good to regenerate the migrations from the current release and rework from there.

Reply:
I have tested the migration. the migration Basically does the following:

1. Renames he model "customsetting" to "oldcustomsetting"
2. Creates the models custombasesetting, jsoncustomsetting, secretcustomsetting.
3. Creates the model customsetting ("should not be an error because now the table custom setting is called oldcustomsetting)
4.  The next migration file, copies the data from oldcustomsetting to the customsetting table
5. entries in the oldcustomsetting gets deleted